frontend/src/entities/jobs/algorithms.tsx

```diff
@@ -71,83 +73,114 @@ export type Hyperparameter = {
     }
-    return context;
-};
 
-const rangeValidator = (
-    fieldName: string,
-    min: number,
-    max: number,
-    isFloat = false,
-    required = true,
-    allowAuto = false
-) => {
-    const validationMessage = `${fieldName} must be ${
-        isFloat ? 'a float' : 'an integer'
-    } value in the range [${min},${max}]${allowAuto ? ' or be \'auto\'' : ''}`;
+    // Create validation message for errors
+    const validationMessage = `${fieldName}${props.required ? ' is required and' : '' } must be ${
+        props.isFloat ? 'a float' : 'an integer'
+    }${rangeMessage}${props.allowAuto ? ' or be \'auto\'' : ''}${props.alternateValues ? ' or be one of the following values: ' + props.alternateValues.map(value => `'${value}'`).join(', ') : ''}`;
+
 
-    const numberContext = z.preprocess(
-        Number,
-        z.number().gte(min, { message: validationMessage }).lte(max, { message: validationMessage })
-    );
+    // Create base number validator
+    let numberContext = z.coerce.number({
+        invalid_type_error: validationMessage,
+    });
 
-    const autoContext = z.literal('auto');
+    // Add min/max check based on min and max inclusions
+    numberContext = props.includeMin ? numberContext.gte(props.min, { message: validationMessage }) : numberContext.gt(props.min, { message: validationMessage });
+    numberContext = props.includeMax ? numberContext.lte(props.max, { message: validationMessage }) : numberContext.lt(props.max, { message: validationMessage });
 
+    // Create a wrapper validator that allows for possible text values
     let wrapperContext: any = numberContext;
 
-    if (allowAuto) {
-        wrapperContext = numberContext.or(autoContext);
+    // Appends auto or other alternative non-number allowed values
+    if (props.allowAuto) {
+        wrapperContext = z.union([wrapperContext, autoContext], {errorMap: () => ({ message: validationMessage})});
+    }
+    if (props.alternateValues) {
+        props.alternateValues.forEach((value) => {
+            wrapperContext = z.union([wrapperContext, z.literal(value)], {errorMap: () => ({ message: validationMessage})});
+        })
     }
 
-    if (!required) {
+    if (!props.required) {
         wrapperContext = wrapperContext.optional();
     }
 
     return wrapperContext;
 };
 
-const positiveIntOrAutoValidator = (fieldName: string, allowIgnore = false) => {
-    const numberContext = z
-        .number()
-        .min(1, { message: positiveIntMessage(fieldName) })
-        .or(z.literal('auto'));
-    if (allowIgnore) {
-        numberContext.or(z.literal('ignore'));
-    }
-
-    const wrapperContext = z.literal('auto').or(z.preprocess(Number, numberContext));
-    if (allowIgnore) {
-        wrapperContext.or(z.literal('ignore'));
-    }
-    return wrapperContext;
+const positiveIntOrAutoValidator = (fieldName: string, alternateValues:string[] = [], allowIgnore = false) => {
```

Comment:
`allowIgnore` wasn't related to the field being required in the original implementation. The `alternatveValues` parameter serves the same purpose of the original allow ignore. It looks like you're primarily calling it with  `alternateValues` of `['ignore','']` which means you're allowing an empty string which would mean it's optional. The original implementation didn't allow for that so I just want to double check that change is intended.
```suggestion
const positiveIntOrAutoValidator = (fieldName: string, alternateValues: string[] = [], required = false) => {
```

Reply:
Understood. I will re-check the fields this applies to and ensure the proper usage.